src/frontend/js/components/DashboardAddressesManagement/DashboardCreateAddressForm.tsx

```diff
@@ -0,0 +1,75 @@
+import { messages as managementMessages } from 'components/AddressesManagement';
+import { useAddressesManagement } from 'components/AddressesManagement/useAddressesManagement';
+import Banner, { BannerType } from 'components/Banner';
+import { Button } from 'components/Button';
+import {
+  AddressFormValues,
+  useDashboardAddressForm,
+} from 'components/DashboardAddressesManagement/useDashboardAddressForm';
+import { DashboardCard } from 'components/DashboardCard';
+import { Spinner } from 'components/Spinner';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+const messages = {
+  header: {
+    id: 'components.DashboardCreateAddressForm.header',
+    description: '',
+    defaultMessage: 'Create an address',
+  },
+  submit: {
+    id: 'components.DashboardCreateAddressForm.submit',
+    description: '',
+    defaultMessage: 'Create',
+  },
+};
+
+interface DashboardCreateAddressFormProps {
+  onDone: Function;
```

Comment:
To keep naming aligned with react-query syntax I will name it `onSuccess` or `onSettled`